processors/base/helpers/__init__.py

```diff
@@ -266,3 +279,47 @@ def safe_prepend(prepend_string, string):
         string = '%s%s' % (prepend_string, string)
 
     return string
+
+
+def get_canonical_location_name(location):
+    """Find the canonical location name according to the
+    passed entry
+
+    Args:
+        location (str): the location to be normalized
+    """
+
+    if location is None:
+        return None
+
+    # Extracted from: https://github.com/datasets/country-codes/blob/master/data/country-codes.csv
+    CSV_PATH = os.path.join(os.path.dirname(__file__), 'data/countries.csv')
+    DISTANCE_SCORER = fuzzywuzzy.fuzz.token_sort_ratio
+
+    clean_string = lambda u: re.sub(u"[^\w\d'\s]+", '', u).lower()
+    try:
+        logger.debug('Location "%s" normalized using ISO-3166 standard', location)
+        return iso3166.countries.get(clean_string(location)).name
+    except KeyError:
+        with open(CSV_PATH, 'r') as csv_file:
+            reader = csv.DictReader(csv_file)
+            current_match = location
+            current_score = float('-inf')
+            for country in reader:
+                relevant_info = [unicode(country[field], encoding='utf-8')
+                                 for field in reader.fieldnames[0:5]]
+
+                cleaned_location = clean_string(location)
+                location_info = [clean_string(location_info) for location_info in relevant_info]
+                _, score = fuzzywuzzy.process.extractOne(cleaned_location, location_info, scorer=DISTANCE_SCORER)
+
+                if score > current_score:
+                    current_match = iso3166.countries.get(country['ISO3166-1-Alpha-3']).name
+                    current_score = score
+
+            if current_score < EDIT_DISTANCE_THRESHOLD:
+                logger.debug('Location "%s" not normalized', location)
+                return location
+
+            logger.debug('Location "%s" normalized using Levenshtein distance', location)
```

Comment:
Log the normalized value as well, as `logger.debug('Location "%s" normalized as "%s"', location, current_match)`